### PR TITLE
cli: authenticate to the Nomad web UI via the CLI

### PR DESCRIPTION
**Pull request by tgross**
Merge PR for the feature branch inclusive of all work except for the web UI work being done by @backspace in #10066. This PR contains the work previously reviewed in #10066 #10091 #10092 #10095 and #10097.

### CODE REVIEW

File: website/content/docs/commands/ui.mdx [hunk 1]
```diff
@@ -22,10 +22,22 @@ will be opened in the default browser.
 An identifier may be provided, in which case the UI will be opened to view the
 details for that object. Supported identifiers are jobs, allocations and nodes.
 
+If ACLs are enabled, the web UI will start in an unauthenticated state and you
+may see a 403 Unauthorized page if anonymous read access is denied. The `nomad
+ui -authenticate` option will exchange your command line client's Nomad ACL
+token for a one-time token to the web UI. That one-time token will be
```
Comment (backspace) [inline website/content/docs/commands/ui.mdx:28]: I see that this was already reviewed in #10097 but is there a word missing here before `to`? Or maybe it should be `for the web UI`?

**Review by backspace — APPROVED** [review]
I had one tiny question about wording and I only looked at documentation aspects but 👍 looks good, thanks!

File: website/content/docs/commands/ui.mdx (hunk 1 above)
Comment (tgross) [reply website/content/docs/commands/ui.mdx:28]: Yeah, that is awkward. What about making it even more explicit:

> The `nomad ui -authenticate` option will exchange your command line client's Nomad ACL token for a one-time token, which is passed to the web UI.

Comment (backspace) [reply website/content/docs/commands/ui.mdx:28]: excellent, thank you!

Comment (tgross) [reply website/content/docs/commands/ui.mdx]: 👍  Done in f4621f2

**Comment by tgross** [comment]
I've rebased this into cleaner commits that can all get rebase-merged to `main`. Will do that once CI is green.

**Pull request merged**